packages/amplify_core/lib/src/category/amplify_push_notifications_category.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+part of amplify_interface;
+
+class PushNotificationsCategory
+    extends AmplifyCategory<PushNotificationsPluginInterface> {
+  @override
+  @nonVirtual
+  Category get category => Category.pushNotifications;
+
+  /// {@template amplify_core.amplify_push_notifications_category.get_permission_status}
+  /// The current Push Notificaiton permission state.
+  ///
+  /// The status can be one of the following:
+  /// 1. notRequested - Android only status to indicate push permissions dialogue has not been requested
+  /// 2. shouldRequestWithRationale - show a rationale message on why the app needs to send push notifications
+  /// 3. granted - end user has granted the request for the app to send push notifications
+  /// 4. denied - end user has denied the request for the app to send push notifications
+  /// {@endtemplate}
+  Future<PushNotificationPermissionRequestStatus> getPermissionStatus() =>
+      defaultPlugin.getPermissionStatus();
+
+  /// {@template amplify_core.amplify_push_notifications_category.request_permissions}
+  /// Request push notifications permissions with options.
+  ///
+  /// On iOS, these control the
+  /// [granular permissions](https://developer.apple.com/documentation/usernotifications/unauthorizationoptions)
+  /// On Android, it has no effect
+  /// It returns true if granted or false if denied.
+  /// {@endtemplate}
+  Future<bool> requestPermissions({
+    bool? alert = true,
+    bool? badge = true,
+    bool? sound = true,
```

Comment:
(Sorry for the late comments)
nit: 
```suggestion
    bool alert = true,
    bool badge = true,
    bool sound = true,
```